cmd/crane/rebase.md

```diff
@@ -0,0 +1,121 @@
+This command rewrites an image's manifest to replace layers in a base image
+with layers in another version of that base image. It does so entirely with API
+calls to the registry, so it doesn't have to download or upload full layer
+blobs, if it can be avoided.
```

Comment:
I think this paragraph should avoid leaking implementation details.  I think we want to get across what it does, and that it does so as efficiently as possible.

Perhaps something like:
> The `rebase` command efficiently rewrites an image to replace the base image it is `FROM` with a new base image.  
> **(insert diagram from below, if possible tweaked to be in terms of FROM?)**
>
> **This is not safe in general,**  but it can be extremely useful for platform providers, e.g. when a vulnerability is discovered in a base layer and they need to patch many thousands (or millions) of applications in a short period of time.  A commonly accepted guideline for rebase-safety is ABI-compatibility, but this is still imperfect in a handful of ways, and the exact contract varies between platform providers.
>
> Rebasing is best suited for when rebuilding is either impossible (no source) or impractical (too much work, too little time).

I think there's value in white lies that let us put things in terms of Dockerfile, even if imprecise.

Reply:
I feel somewhat beaten over the head by the vulnerability use case.  I think we should give it as the canonical example early on and focus the rest as much as possible on just rebasing less the specific use case.